core/src/main/java/com/sequenceiq/cloudbreak/service/stopstart/RecoveryCandidateCollectionService.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.cloudbreak.service.stopstart;
+
+import static com.sequenceiq.cloudbreak.cloud.model.HostName.hostName;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.HostName;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterHealthService;
+import com.sequenceiq.cloudbreak.cluster.status.DetailedHostStatuses;
+import com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil;
+import com.sequenceiq.cloudbreak.converter.CloudInstanceIdToInstanceMetaDataConverter;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.RuntimeVersionService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+
+@Service
+public class RecoveryCandidateCollectionService {
```

Comment:
UT is missing

Reply:
Will add UT for this, although the methods here are tested via the upscale and downscale recovery handlers

Reply:
UTs of handler should mock method calls of this service, and service should have different UT test. That is the point of UT, to test only the given class (=unit) and nothing else.

Reply:
Sure, thanks, will refactor them

Reply:
Thanks!